src/condor_scripts/condor_upgrade_check

```diff
@@ -1,555 +1,151 @@
-                    bracket_cnt = 0
-                    prev_char = None
-                    pos = 0
-                    # Manually check sequence line for hyphens in non common cases
-                    for char in sequence:
-                        if char == "[":
-                            bracket_cnt += 1
-                        elif char == "]":
-                            bracket_cnt -= 1
-                        elif char == "-" and bracket_cnt > 0:
-                            # Ignore correct/common cases
-                            if sequence[pos + 1] == "]":
-                                # Hyphen at end of pcre statement
-                                pass
-                            elif prev_char == "A" and sequence[pos + 1] == "Z":
-                                # A-Z
-                                pass
-                            elif prev_char == "a" and sequence[pos + 1] == "z":
-                                # a-z
-                                pass
-                            elif prev_char == "0" and sequence[pos + 1] == "9":
-                                # 0-9
-                                pass
-                            else:
-                                # Not a normal case so output
-                                num_errors += 1
-                                format_print(
-                                    f"Possibly Invalid: {filename}|line-{line_num}: '{line}'",
-                                    offset=12,
-                                )
-                                break
-                        prev_char = char
-                        pos += 1
-                if not has_cmd and is_el7:
-                    # No 'pcre2grep' cmd and EL7 so manually check for known incompatibility
-                    for char in PCRE2_POSIX_CHARS:
-                        if char in line:
-                            pos = line.find(char)
-                            if line[pos + len(char)] == "-":
-                                num_errors += 1
-                                format_print(
-                                    f"Invalid: {filename}|line-{line_num}: '{line}'",
-                                    offset=12,
-                                )
-                                break
-    except IOError:
-        num_errors += 1
-        format_print(f"ERROR: Failed to open {filename}", offset=12, err=True)
-    return num_errors
-
-
-def check_pcre2():
-    """Check map files for known PCRE2 breakages"""
-    is_el7 = False
-    has_pcre2_cmd = False
-    # Check if platform is EL7
-    if platform.system() == "Linux" and "el7" in platform.release():
-        is_el7 = True
-    format_print("Checking for incompatibilities with PCRE2.", newline=True)
-    format_print("Known PCRE2 incompatibilities:", offset=8)
+
+# ===============================Incompatibility Check Functions===============================
+def warn_dagman_env(args):
```

Comment:
Kind of a nitpick, but since `args` is never used here, consider using the standard catch-alls so that passing in a positional argument and/or keyword argument isn't required, but the function also doesn't except if they are passed in:
```suggestion
def warn_dagman_env(*args, **kwargs):
```